lib/ProxyFactory.php

```diff
@@ -95,6 +91,9 @@ public function createProxyForNode($fromDocument, NodeInterface $targetNode)
             return $document;
         }
 
+        $eventDispatcher = $this->dispatcher;
+        $registry = $this->registry;
```

Comment:
Is there a reason you are doing this assignment? Why not just using `$this->dispatcher` and `$this->registry`?